mlflow/tracking/_tracking_service/client.py

```diff
@@ -80,7 +81,38 @@ def get_metric_history(self, run_id, key):
 
         :return: A list of :py:class:`mlflow.entities.Metric` entities if logged, else empty list
         """
-        return self.store.get_metric_history(run_id=run_id, metric_key=key)
+
+        # NB: Paginated query support is currently only available for the RestStore backend.
+        # FileStore and SQLAlchemy store do not provide support for paginated queries and will
+        # raise an MlflowException if the `page_token` argument is not None when calling this
+        # API for a continuation query.
+
+        max_results = GET_METRIC_HISTORY_MAX_RESULTS
+
+        history = self.store.get_metric_history(
+            run_id=run_id, metric_key=key, max_results=max_results, page_token=None
+        )
+        if not isinstance(history, PagedList):
+            # Backwards compatible return if the server doesn't support paginated queries
+            return history
+        else:
+            if history.token is None:
+                # If the request is small enough to be fulfilled with a single query, return it.
+                return history.to_list()
+            else:
+                metric_collection = history.to_list()
+                # Continue issuing queries to the backend store to retrieve all pages of
+                # metric history.
+                while history.page_token is not None:
+                    history = self.store.get_metric_history(
+                        run_id=run_id,
+                        metric_key=key,
+                        max_results=max_results,
+                        page_token=history.page_token,
+                    )
+                    metric_collection.extend(history.to_list())
+                # Return the flattened list of paginated metric history entries
+                return [metrics for page in metric_collection for metrics in page]
```

Comment:
```suggestion
                return metric_collection
```

`metric_collection` is already flattened, right?

Reply:
yep with the change to `.extend` and direct list assignment. I'll update this :)